diff --git a/dist/web-worker/leader-worker/make-leader-worker.js b/dist/web-worker/leader-worker/make-leader-worker.js
index 4209c65402d95301e6e4271d80d34022c5e9f8b4..d2677189bcbd7fb0145a90a1c1ed6b73280c9f55 100644
--- a/dist/web-worker/leader-worker/make-leader-worker.js
+++ b/dist/web-worker/leader-worker/make-leader-worker.js
@@ -50,6 +50,7 @@ const makeMessagePortRpcServerProtocol = (port) => Layer.effect(RpcServer.Protoc
         supportsTransferables: true,
         supportsSpanPropagation: true,
         supportsNotifications: true,
+        codecFor: Schema.toCodecJson,
         run: (writeRequest) => Effect.gen(function* () {
             const context = yield* Effect.context();
             const runFork = Effect.runForkWith(context);
diff --git a/src/web-worker/leader-worker/make-leader-worker.ts b/src/web-worker/leader-worker/make-leader-worker.ts
index ffd415759f8fb9fb5377deb596d02bf18f893d5a..fd38cbabc86305a7788a2af96362e2bca058bdd4 100644
--- a/src/web-worker/leader-worker/make-leader-worker.ts
+++ b/src/web-worker/leader-worker/make-leader-worker.ts
@@ -138,6 +138,7 @@ const makeMessagePortRpcServerProtocol = (port: MessagePort): Layer.Layer<RpcSer
         supportsTransferables: true,
         supportsSpanPropagation: true,
         supportsNotifications: true,
+        codecFor: Schema.toCodecJson,
         run: (writeRequest) =>
           Effect.gen(function* () {
             const context = yield* Effect.context<never>()
